crates/tx5-connection/src/config.rs

```diff
@@ -57,21 +57,22 @@ pub struct IceServers {
     pub urls: Vec<String>,
 
     /// The username to use for authentication.
-    #[serde(default)]
+    #[serde(default, skip_serializing_if = "Option::is_none")]
     pub username: Option<String>,
 
     /// The credential to use for authentication.
-    #[serde(default)]
+    #[serde(default, skip_serializing_if = "Option::is_none")]
     pub credential: Option<String>,
 
     /// The credential type to use for authentication.
-    #[serde(default)]
+    #[serde(default, skip_serializing_if = "Option::is_none")]
     pub credential_type: Option<CredentialType>,
 }
 
 /// ICE transport policy.
 #[derive(Debug, Clone, Default, Serialize, Deserialize)]
 #[cfg_attr(feature = "schema", derive(schemars::JsonSchema))]
+#[serde(rename_all = "camelCase")]
```

Comment:
It's unclear whether datachannel actually uses this value, but for go-pion the case needed fixing https://github.com/pion/webrtc/blob/master/icetransportpolicy.go#L27-L30

Reply:
We would not necessarily, because it doesn't need a valid ICE configuration. Everything works fine on a local network but falls apart when you need to know your public IP